src/legacy/core_plugins/kibana/public/visualize/plugin.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// @ts-ignore
+import { SavedVisualizations } from './saved_visualizations/saved_visualizations';
+
+import { Dependencies, Management, Ui, Visualize } from '../../plugin_dependencies';
+
+// TODO: import real type from core once available
+interface CoreStart {
+  [key: string]: any;
+}
+
+export interface VisualizeStartDependencies extends Dependencies {
+  management: Pick<Management, 'savedObjectManagementRegistry'>;
+  ui: Pick<
+    Ui,
+    | 'createLegacyClass'
+    | 'savedObjectLoader'
+    | 'savedObjectProvider'
+    | 'savedObjectsClientProvider'
+    | 'uiModules'
+  >;
+  visualize: Pick<Visualize, 'updateOldState' | 'visProvider' | 'visTypesRegistryProvider'>;
+}
+
+export class VisualizePlugin {
+  private readonly savedVisualizations: SavedVisualizations;
+
+  constructor() {
+    this.savedVisualizations = new SavedVisualizations();
```

Comment:
nit: This is effectively a service, so in core we'd call this class `SavedVisualizationService`. This is entirely convention though, and the name of your service class is an implementation detail of your plugin, so feel free to ignore this information.

See https://github.com/elastic/kibana/blob/master/src/core/public/core_system.ts#L61-L101 for examples in core.